blog/2022-07-05-paralus-quickstart-gke/index.md

```diff
@@ -114,6 +115,23 @@ Org Admin signup URL:  http://console.chartexample.com/self-service/recovery?flo
 
 ```
 
+#### Recovering Password Reset Link
+
+The password recovery link generated while deploying Paralus is valid for `10 minutes`. For any reason if the link is expired, you can use the following code snippet to generate the recovery link for any user.
+
+> **Note:** Provide the email id of the user whose password you wish to retrieve. If you've set a username and password for the postgresql database, please replace `admindbpassword` and `admindbuser` with your values.
+
+```bash
+export RELEASE_NAME=<HELM_RELEASE_NAME>
+export RUSER=<USER_ADMIN_EMAIL>
+
+kubectl exec -it "$RELEASE_NAME-postgresql-0" -- bash \
+  -c "PGPASSWORD=admindbpassword psql -h localhost -U admindbuser admindb \
+-c \"select id from identities where traits->>'email' = '$RUSER' limit 1;\" -tA \
+| xargs -I{} curl -X POST http://$RELEASE_NAME-kratos-admin/recovery/link \
+-H 'Content-Type: application/json' -d '{\"expires_in\":\"10m\",\"identity_id\":\"{}\"}'"
```

Comment:
```suggestion
export RELEASE_NAME=<HELM_RELEASE_NAME>
export RUSER=<USER_ADMIN_EMAIL>
export RNAMESPCE=<NAMESPCE>

kubectl exec -it "$RELEASE_NAME-postgresql-0" -n "$RNAMESPACE" -- bash \
  -c "PGPASSWORD=admindbpassword psql -h localhost -U admindbuser admindb \
-c \"select id from identities where traits->>'email' = '$RUSER' limit 1;\" -tA \
| xargs -I{} curl -X POST http://$RELEASE_NAME-kratos-admin/recovery/link \
-H 'Content-Type: application/json' -d '{\"expires_in\":\"10m\",\"identity_id\":\"{}\"}'"
```

Also, this is only valid if one uses the postgres bundled with paralus and not if they are using something external. If using external, after fetching user id from the db corresponding to the admin user, one can use
```
curl -X POST http://$RELEASE_NAME-kratos-admin/recovery/link \
-H 'Content-Type: application/json' -d '{"expires_in":"10m","identity_id":"<ADMIN_USER_ID>"}'
```

Reply:
@meain What do you mean by 'something external' and not the bundled postgres? How does a user configure it, have we mentioned this anywhere?

Reply:
As in instead of setting `postgres: true` they provide the username/password to an [external postgres](https://github.com/paralus/helm-charts/blob/a45ec2e981530c90c2cf6f27a8f8aae98607be55/charts/ztka/values.yaml#L210=) instance that we can use.